monkey/monkey_island/cc/server_utils/setup_flask.py

```diff
@@ -0,0 +1,24 @@
+import json
+import secrets
+from pathlib import Path
+from typing import Any, Dict
+
+from common.utils.file_utils import open_new_securely_permissioned_file
+
+SECRET_FILE_NAME = ".flask_security_configuration.json"
+
+
+def generate_flask_security_configuration(data_dir: Path) -> Dict[str, Any]:
+    SECRET_FILE_PATH = str(data_dir / SECRET_FILE_NAME)
+    try:
+        with open(SECRET_FILE_PATH, "r") as secret_file:
+            return json.load(secret_file)
+    except FileNotFoundError:
+        with open_new_securely_permissioned_file(SECRET_FILE_PATH, "w") as secret_file:
+            secret_key = secrets.token_urlsafe(32)
+            password_salt = secrets.SystemRandom().getrandbits(128)
```

Comment:
I'm not sure if we need `SECURITY_PASSWORD_SALT` at all. Read [this thread](https://github.com/mattupstate/flask-security/issues/135) that discusses it.

Reply:
Seems like overkill to me going by [this comment](https://github.com/mattupstate/flask-security/issues/135#issuecomment-340542748) which says that the only time this will help is if someone gets access to the database (which is already pretty secure) and the salt used in the password hash.

It's just another file with a secret to worry about.

Reply:
IIRC if someone somehow figures out how the salt is being generated (if "random" turns out to be predictable in salt generation) there's still a chance that this "pepper" is not found and thus hashes can't be cracked. I think if we're storing the secret key might as well add the secret salt